jib-core/src/main/java/com/google/cloud/tools/jib/cache/DefaultCachedLayer.java

```diff
@@ -68,44 +69,47 @@ CacheEntry build() {
   }
 
   /**
-   * Creates a new {@link Builder} for a {@link CacheEntry}.
+   * Creates a new {@link Builder} for a {@link CachedLayer}.
    *
    * @return the new {@link Builder}
    */
   static Builder builder() {
     return new Builder();
   }
 
-  private final DescriptorDigest layerDigest;
   private final DescriptorDigest layerDiffId;
-  private final long layerSize;
+  private final BlobDescriptor blobDescriptor;
   private final Blob layerBlob;
 
-  private DefaultCacheEntry(
+  private DefaultCachedLayer(
       DescriptorDigest layerDigest, DescriptorDigest layerDiffId, long layerSize, Blob layerBlob) {
-    this.layerDigest = layerDigest;
     this.layerDiffId = layerDiffId;
-    this.layerSize = layerSize;
     this.layerBlob = layerBlob;
+    this.blobDescriptor = new BlobDescriptor(layerSize, layerDigest);
   }
 
   @Override
-  public DescriptorDigest getLayerDigest() {
-    return layerDigest;
+  public DescriptorDigest getDigest() {
+    return blobDescriptor.getDigest();
   }
 
   @Override
-  public DescriptorDigest getLayerDiffId() {
+  public DescriptorDigest getDiffId() {
     return layerDiffId;
   }
 
   @Override
-  public long getLayerSize() {
-    return layerSize;
+  public long getSize() {
+    return blobDescriptor.getSize();
   }
 
   @Override
-  public Blob getLayerBlob() {
+  public Blob getBlob() {
     return layerBlob;
   }
+
+  @Override
+  public BlobDescriptor getBlobDescriptor() {
```

Comment:
Hmm, can we remove this from Layer interface?

Reply:
I tried and it looks like there may end up being lots of changes, maybe not worth it for now.

Reply:
Okay let's add a TODO to remove this later then.